src/Catalyst.TestUtils/FileSystemBasedTest.cs

```diff
@@ -0,0 +1,147 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using Catalyst.Abstractions.FileSystem;
+using Catalyst.Core.Lib.Config;
+using Catalyst.Core.Lib.FileSystem;
+using Catalyst.Protocol.Network;
+using Dawn;
+using FluentAssertions;
+using NSubstitute;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.TestUtils
+{
+    /// <inheritdoc />
+    /// <summary>
+    ///     A base test class that can be used to offer inheriting tests a folder on which
+    ///     to create files, logs, etc.
+    /// </summary>
+    [Trait(Traits.TestType, Traits.IntegrationTest)]
+    public class FileSystemBasedTest : IDisposable
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fix this implementation of 'IDisposable' to conform to the dispose pattern.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)